protocols/gossipsub/src/transform.rs

```diff
@@ -41,11 +43,7 @@ pub trait DataTransform {
 
     /// Takes the data to be published (a topic and associated data) transforms the data. The
     /// transformed data will then be used to create a [`crate::RawMessage`] to be sent to peers.
-    fn outbound_transform(
```

Comment:
This is a public trait so we should mention the signature chance in the changelog.